### PR TITLE
Changing to Material UI and isolating Bootstrap

### LINKED ISSUE #532: Make FROG graph in Teacher View span the width of the screen
Any reason why not? (I guess we could have a cut off if it's a very large screen and a very small graph)... Also rethink a bit how we locate the information about dashboard, students active, graph, etc.

### PR DESCRIPTION
**Pull request by houshuang**
Replaces #746 

@aperritano OK, big progress. I followed some of the guidelines in the Stackoverflow article I linked to above, and was able to create a Bootstrap file which only applies its rules to content within <div class='bootstrap'>... </div>, and it seems to work well - when I start FROG like this, the graph editor is broken, but adding the class makes the graph editor look fine.

There are still a number of issues:
- in the future, please make sure you submit clean PRs, this PR also had ac-ck-student, which is a completely different issue from changing the core UI to Material
- you also removed all the text in the README.md, renamed FROG to CK etc. Again, don't make this part of a PR - if you want to make some part of the FROG UI configurable (like the name of the platform), we can consider that, but don't hardcode changes - makes it much harder to merge.
- make sure to run Prettier and ESLint etc

I have uploaded a branch stiantony, which has your changes, minus ac-ck-student, merged the latest develop, run prettier, add isolated-bootstrap, and apply to it to the graph editor and also ac-form (just to see the difference in react-jsonschema-form with and without Bootstrap).

I would suggest you continue working on this branch and make sure it's solid before we can merge:
- the React Mosaic view looks very bad now, not sure if it requires Bootstrap or we messed up something else (easy to see in Preview mode adding multiple students)
- the graph editor works perfectly now, however the alignment is off - it needs to be pushed down a bit, and also the pixel alignment in the graph editor (probably in frog/imports/ui/GraphEditor/store/activity.js:184 (@get y)) needs to be updated so that the rename fields are aligned etc.
- the top bar does not show graph editor active, maybe because the url also has the graph id in it?
- the create session/switch session in teacher view (sessions) should be updated to Material
- make sure all tests pass (now that the branch is on FROG repo, you can see the CI results whenever you push)

Once you fix these and are happy with the PR (also review the diff on Github and make sure only relevant changes are included, tests pass etc), I'm happy to look at this again and get it merged quickly.

Closes #532

### CODE REVIEW

**Comment by aperritano** [comment]
Good job on making progress toward this issue. 

When you checked in the changes it working without error? After a clean checkout I'm getting: 

1)
Error: Cannot find module '../lib/utils/unsupported.js'
    at Function.Module._resolveFilename (module.js:542:15)
    at Function.Module._load (module.js:472:25)
    at Module.require (module.js:585:17)
    at require (internal/module.js:11:18)
    at /usr/local/lib/node_modules/npm/bin/npm-cli.js:19:21
    at Object.<anonymous> (/usr/local/lib/node_modules/npm/bin/npm-cli.js:92:3)
    at Module._compile (module.js:641:30)
    at Object.Module._extensions..js (module.js:652:10)
    at Module.load (module.js:560:32)
    at tryModuleLoad (module.js:503:12)
module.js:544
    throw err;

2)--------

caught TypeError: Cannot read property 'to' of undefined
    at new TopBar (TopBar.js:55)
    at constructClassInstance (modules.js?hash=2e31dfa2a33b13171479ff28d2400dc98962bba4:21817)
    at updateClassComponent (modules.js?hash=2e31dfa2a33b13171479ff28d2400dc98962bba4:23301)
    at beginWork (modules.js?hash=2e31dfa2a33b13171479ff28d2400dc98962bba4:23687)
    at performUnitOfWork (modules.js?hash=2e31dfa2a33b13171479ff28d2400dc98962bba4:25686)
    at workLoop (modules.js?hash=2e31dfa2a33b13171479ff28d2400dc98962bba4:25750)
    at HTMLUnknownElement.callCallback (modules.js?hash=2e31dfa2a33b13171479ff28d2400dc98962bba4:16004)
    at Object.invokeGuardedCallbackDev (modules.js?hash=2e31dfa2a33b13171479ff28d2400dc98962bba4:16043)
    at invokeGuardedCallback (modules.js?hash=2e31dfa2a33b13171479ff28d2400dc98962bba4:15900)
    at renderRoot (modules.js?hash=2e31dfa2a33b13171479ff28d2400dc98962bba4:25828)
TopBar @ TopBar.js:55
constructClassInstance @ modules.js?hash=2e31dfa2a33b13171479ff28d2400dc98962bba4:21817
updateClassComponent @ modules.js?hash=2e31dfa2a33b13171479ff28d2400dc98962bba4:23301
beginWork @ modules.js?hash=2e31dfa2a33b13171479ff28d2400dc98962bba4:23687
performUnitOfWork @ modules.js?hash=2e31dfa2a33b13171479ff28d2400dc98962bba4:25686
workLoop @ modules.js?hash=2e31dfa2a33b13171479ff28d2400dc98962bba4:25750
callCallback @ modules.js?hash=2e31dfa2a33b13171479ff28d2400dc98962bba4:16004
invokeGuardedCallbackDev @ modules.js?hash=2e31dfa2a33b13171479ff28d2400dc98962bba4:16043
invokeGuardedCallback @ modules.js?hash=2e31dfa2a33b13171479ff28d2400dc98962bba4:15900
renderRoot @ modules.js?hash=2e31dfa2a33b13171479ff28d2400dc98962bba4:25828
performWorkOnRoot @ modules.js?hash=2e31dfa2a33b13171479ff28d2400dc98962bba4:26476
performWork @ modules.js?hash=2e31dfa2a33b13171479ff28d2400dc98962bba4:26429
requestWork @ modules.js?hash=2e31dfa2a33b13171479ff28d2400dc98962bba4:26340
scheduleWorkImpl @ modules.js?hash=2e31dfa2a33b13171479ff28d2400dc98962bba4:26194
scheduleWork @ modules.js?hash=2e31dfa2a33b13171479ff28d2400dc98962bba4:26151
enqueueForceUpdate @ modules.js?hash=2e31dfa2a33b13171479ff28d2400dc98962bba4:21712
Component.forceUpdate @ modules.js?hash=2e31dfa2a33b13171479ff28d2400dc98962bba4:12374
(anonymous) @ ReactMeteorData.jsx:79
Tracker.Computation._compute @ tracker.js?hash=997515fa2d5b0530ba07741da556c4b36963ef3b:339
Tracker.Computation._recompute @ tracker.js?hash=997515fa2d5b0530ba07741da556c4b36963ef3b:358
Tracker._runFlush @ tracker.js?hash=997515fa2d5b0530ba07741da556c4b36963ef3b:532
onGlobalMessage @ meteor.js?hash=cbcc712d51de4298c275e8dcf25c66c29914f19a:448
postMessage (async)
setImmediate @ meteor.js?hash=cbcc712d51de4298c275e8dcf25c66c29914f19a:468
requireFlush @ tracker.js?hash=997515fa2d5b0530ba07741da556c4b36963ef3b:151
Tracker.Computation.invalidate @ tracker.js?hash=997515fa2d5b0530ba07741da556c4b36963ef3b:292
Tracker.Dependency.changed @ tracker.js?hash=997515fa2d5b0530ba07741da556c4b36963ef3b:459
(anonymous) @ livedata_connection.js:1509
(anonymous) @ livedata_connection.js:1364
_.each._.forEach @ underscore.js?hash=cde485f60699ff9aced3305f70189e39c665183c:149
_runAfterUpdateCallbacks @ livedata_connection.js:1363
_performWrites @ livedata_connection.js:1353
_flushBufferedWrites @ livedata_connection.js:1316
_livedata_data @ livedata_connection.js:1284
onMessage @ livedata_connection.js:274
(anonymous) @ stream_client_sockjs.js:174
_.each._.forEach @ underscore.js?hash=cde485f60699ff9aced3305f70189e39c665183c:149
self.socket.onmessage @ stream_client_sockjs.js:173
REventTarget.dispatchEvent @ sockjs-0.3.4.js:87
SockJS._dispatchMessage @ sockjs-0.3.4.js:1072
SockJS._didMessage @ sockjs-0.3.4.js:1130
that.ws.onmessage @ sockjs-0.3.4.js:1277
14:06:06.420 modules.js?hash=2e31dfa2a33b13171479ff28d2400dc98962bba4:25209 The above error occurred in the <TopBar> component:
    in TopBar (created by withStyles(TopBar))
    in withStyles(TopBar) (created by Route)
    in Route (created by withRouter(withStyles(TopBar)))
    in withRouter(withStyles(TopBar)) (created by WithTopBar)
    in div (created by WithTopBar)
    in WithTopBar (created by Route)
    in Route (created by TeacherContainer)
    in Switch (created by TeacherContainer)
    in div (created by TeacherContainer)
    in TeacherContainer (created by ReactMeteorDataComponent)
    in ReactMeteorDataComponent (created by LoadableComponent)
    in LoadableComponent (created by Route)
    in Route (created by RawRouter)
    in RawRouter (created by Route)
    in Route (created by withRouter(RawRouter))
    in withRouter(RawRouter) (created by Route)
    in Route
    in Switch
    in div
    in Router (created by BrowserRouter)
    in BrowserRouter
    in _ErrorBoundary
    in Unknown
    in MuiThemeProvider (created by MuiThemeProviderWrapper)
    in MuiThemeProviderWrapper

React will try to recreate this component tree from scratch using the error boundary you provided, _ErrorBoundary.

**Comment by lfaucon** [comment]
@aperritano I reported this second error in my comment on your original PR #746

**Comment by houshuang** [comment]
@aperritano You need to specify /teacher in the URL. This is something you should fix, but when you do, it works.

screen shot 2018-01-16 at 15 25 54

**Comment by houshuang** [comment]
@aperritano I moved the className=bootstrap from the whole graph editor to just the actual graph component, which took away the Bootstrap collision in the title bar.

screen shot 2018-01-19 at 09 24 44

Of course, now everything else looks crappy, but this should be easy to fix. Eventually this page will be completely redesigned, but for now we need it functional. If you can clean up the rest - put the menus into Material format, adjust the CSS to avoid the top bar overlapping, etc, it would be great to get this merged. I guess the most important thing for now is to have functional (if not pretty) UI for the core, and to avoid any Bootstrap collisions with your activity, which will be the next code to review. 

Let me know if you hit any snags.

**Comment by houshuang** [comment]
# Issues with latest version (266cfc2):
- tabs on top of preview window (example list)
- the React Mosaic works but looks less nice
Compare before:
screen shot 2018-01-23 at 08 18 11

After:
screen shot 2018-01-23 at 08 18 20

- ac-prox looks bad (easy fix would be to wrap in Bootstrap)
screen shot 2018-01-23 at 09 31 44

- ac-image, the button to close the modal-pop up is broken
- ac-classify, star does not show up on selection
- ac-induction, tab system needs to be converted
- ac-brainstorm, list and form look ugly, need upvote/downvote buttons
- ac-quiz, needs a tiny bit of css to keep question and selector on the same line
- Sessions view should get rid of the scroll bar of the right of graph editor, and fix Create Session and Switch Session to dropdowns

screen shot 2018-01-23 at 09 51 07

- Major problem with the student view, first, with a single activity, the background is somehow red
screen shot 2018-01-23 at 09 46 30

- And with two activities, they disappear! I guess this is some interaction between React-Mosaic and Material or something like that... You can see here that the two activities are rendered to DOM, but they're not visible
screen shot 2018-01-23 at 09 50 28

# In conclusion
This looks like a lot, but we're getting there for sure. I think that for all the activities that have problems, the quickest short-term solution is just wrapping the individual ActivityRunners in bootstrap for now - pending a redesign to a proper nice design. I can take care of that. 

If you could focus on the UI problems with the core FROG system that I listed above:
- fix tabs in Preview
- fix pull-downs in Session view (these two will also be good models for us in terms of how this should be done with MUI)
- make React Mosaic look good in preview, and work at all in student view (highest priority! :))
- avoid red background in student view

**Comment by houshuang** [comment]
Hi @aperritano - can just have a quick look now, preparing to travel to a conference, but it's beginning to look very interesting.

Remember to preventDefault in the title and time changer, when I click enter, it reloads the page !:) Will look more and let you know when I get time.

**Comment by houshuang** [comment]
Good progress, biggest issue is still that student view is not working at all - this is blocking! With single activity, screen is still red for some reason. With more than one activity - nothing is shown...

**Comment by lfaucon** [comment]
@aperritano Setting height of the body to 100% seems to fix the issue Stian mentioned on the StudentView. Does it conflict with something else that you have changed? 

It looks very good to me now.

**Comment by aperritano** [comment]
If there are no conflicts with teacher view then its good to go.

**Comment by prastut** [comment]
Closed in favour of #900
